src/WebJobs.Script.WebHost/Filters/RequiresRunningHostAttribute.cs

```diff
@@ -61,17 +64,24 @@ public RunningHostCheckAttribute(IScriptHostManager hostManager)
 
             public override async Task OnActionExecutionAsync(ActionExecutingContext actionContext, ActionExecutionDelegate next)
             {
-                // If the host is not ready, we'll wait a bit for it to initialize.
-                // This might happen if http requests come in while the host is starting
-                // up for the first time, or if it is restarting.
-                bool hostReady = await _hostManager.DelayUntilHostReady(TimeoutSeconds, PollingIntervalMilliseconds);
-
-                if (!hostReady)
+                if (_hostManager.State == ScriptHostState.Offline)
                 {
-                    throw new HttpException(HttpStatusCode.ServiceUnavailable, "Function host is not running.");
+                    await HostAvailabilityCheckMiddleware.ReturnOfflineResponse(actionContext.HttpContext, _applicationHostOptions.CurrentValue.ScriptPath);
```

Comment:
Need to confirm that the portal is set up to handle this 503 offline response - it should be